packs/datadog_agent/templates/pack.nomad.tpl

```diff
@@ -0,0 +1,146 @@
+    }
+
+    [[- $job_tags := .my.job_tags -]]
+    [[- $service_name := .my.service_name_prefix -]]
+    [[- $service_provider := .my.service_provider -]]
+    [[/* iterate over `$ports` to map Services */]]
+    [[ range $name, $port := $ports ]]
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/service
+    service {
+      name     = "[[ $service_name | replace "_" "-" | trunc 20 ]]-[[ $name | replace "_" "-" | trunc 43 ]]"
+      tags     = [[ $job_tags | toJson ]]
+      port     = [[ $port.port ]]
+      provider = "[[ $service_provider ]]"
+
+      [[- /* specifically ignore IPC, GUI, and DogStatsD ports, as the check require unsupported strategies */]]
+      [[ if not (or (eq $port.port 5001) (eq $port.port 5002) (eq $port.port 8125)) ]]
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/check
+      check {
+        name            = "[[ $name ]]"
+        type            = "[[ $port.type ]]"
+        [[- if eq $port.type "http" ]]
+        path            = "[[ $port.path ]]"
+        protocol        = "[[ $port.protocol ]]"
+        [[- end ]]
+        interval        = "[[ $port.check_interval ]]"
+        timeout         = "[[ $port.check_timeout ]]"
+      }
+      [[ end ]]
+    }
+    [[ end ]]
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/restart
+    restart {
+      attempts = [[ .my.restart_logic.attempts ]]
+      interval = "[[ .my.restart_logic.interval ]]"
+      delay    = "[[ .my.restart_logic.delay ]]"
+      mode     = "[[ .my.restart_logic.mode ]]"
+    }
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/volume
+    [[/* iterate over `var.volumes` to create Volumes */]]
+    [[- range $index, $mount := .my.volumes ]]
+    volume "[[ $mount.name ]]" {
+      source    = "[[ $mount.name ]]"
+      type      = "[[ $mount.type ]]"
+      read_only = [[ $mount.read_only ]]
+    }
+    [[ end ]]
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/task
+    task "[[ .my.task_name ]]" {
+      # see https://developer.hashicorp.com/nomad/docs/drivers
+      driver = "[[ .my.driver ]]"
+
+      # see https://developer.hashicorp.com/nomad/docs/drivers/raw_exec
+      # and https://developer.hashicorp.com/nomad/docs/drivers/exec
+      config {
+        # see https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6v7
+        command = "datadog-agent"
+        args    = ["run"]
+      }
+
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/env
+      env {
+        [[- template "configuration" . ]]
+      }
+
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/template
+      template {
+        change_mode = "restart"
+
+        # render template with sensitive data
+        # see https://app.datadoghq.com/account/settings
+        data = <<EOF
+          {{- with nomadVar "nomad/jobs/[[ .my.job_name ]]" -}}
+          DD_API_KEY = "{{ .api_key }}"
```

Comment:
This should be documented in the README as a prerequisite

Reply:
Good one - fixed!